examples/run_demo_ursula_fleet.py

```diff
@@ -52,23 +51,29 @@ def spin_up_federated_ursulas(quantity: int = FLEET_POPULATION):
             rest_port=port,
             seed_nodes=[sage.seed_node_metadata()],
             start_learning_now=True,
-            db_filepath=f"{Path(APP_DIR.user_cache_dir) / port}.db",
+            db_filepath=f"{USER_CACHE / port}.db",
         )
         ursulas.append(u)
+
     for u in ursulas:
         deployer = u.get_deployer()
         deployer.addServices()
         deployer.catalogServers(deployer.hendrix)
         deployer.start()
         print(f"{u}: {deployer._listening_message()}")
+
     try:
-        reactor.run()  # GO!
```

Comment:
I'm slightly confused here - without the reactor running, how do the Ursulas startup services?  Are you certain they were not still running from a previous test?

Reply:
Hmmm... ok that's a good question. Let me double check. 
If I use reactor.run... how can I shut the ursulas down from the command line?

Reply:
ok.. you are right... so, `reactor.run` is problematic, because the user loses control of the terminal. The only way to shut the program down is to kill the terminal, or to kill the python program. (In a separate terminal)

Even if this were an acceptable way to stop this script, killing the program skips the finally block.

Do you know how I can call `reactor.run` in a way where the user still has control of the terminal, and how I would shut the Ursulas down gracefully?

Reply:
> ok.. you are right... so, reactor.run is problematic, because the user loses control of the terminal.

In your view, why is this problematic?

> Even if this were an acceptable way to stop this script, killing the program skips the finally block.

`KeyboardInterrupt` skips the finally block?


> Do you know how I can call reactor.run in a way where the user still has control of the terminal, and how I would shut the Ursulas down gracefully?

Yes, it's possible to use a twisted `LineReceiever` for this purpose, but I don't think it's worth the complexity just for a demo.

Reply:
I mean... I can't stop the demo from running in the shell once it starts.
So as far as I can tell.. there is no way to keyboard interrupt.
I would have to kill the terminal... and the finally block doesn't fire when I do

Reply:
> So as far as I can tell.. there is no way to keyboard interrupt.

What happens when you CTRL+C ?

Reply:
nothing... I cannot stop this process.
Can you?

Reply:
> Can you?

Yes, but it's not consistent.  CTRL+C early on in the runtime seems to stop normally.  However if I wait about ten or twenty seconds for services to fully startup, that does not work anymore, so I end up halting the process with a core dump (CTRL+\ on ubuntu).

This is not expected behavior, if this process does not respond to a `KeyboardInterrupt`, it's most likely an indication that there is an unhandled exception or missing twisted error handler